src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -50,27 +116,341 @@ def execute(self, reqStatus):
+                            number of processed workflows
+                            number of archived workflows
+        """
+        # NOTE: The Input Cleanup, the Block Level Cleanup and the Archival
+        #       Pipelines are executed sequentially in the above order.
+        #       This way we assure ourselves that we archive only workflows
+        #       that have accomplished the needed cleanup
+
+        clean_num_requests = 0
+        total_num_requests = 0
+
+        # Call the workflow dispatcher:
+        for reqName, req in reqRecords.items():
+            wflow = MSRuleCleanerWflow(reqRecords[reqName])
+            self._dispatchWflow(wflow)
+            msg = "\n----------------------------------------------------------"
+            msg += "\nMSRuleCleanerWflow: %s"
+            msg += "\n----------------------------------------------------------"
+            self.logger.debug(msg, pformat(wflow))
+            total_num_requests += 1
+            if self._checkClean(wflow):
+                clean_num_requests += 1
+
+        # Report the counters:
+        for pline in self.cleanuplines:
+            msg = "Workflows cleaned by pipeline: %s: %d"
+            self.logger.info(msg, pline.name, self.wfCounters['cleaned'][pline.name])
+        archived_num_requests = self.wfCounters['archived']
+        msg = "Workflows archived: %d"
+        self.logger.info(msg, self.wfCounters['archived'])
+        return total_num_requests, clean_num_requests, archived_num_requests
+
+    def _dispatchWflow(self, wflow, *args, **kwargs):
+        """
+        A function intended to dispatch a workflow (e.g based on its status)
+        through one or more functional pipelines in case there is some more
+        complicated logic involved in the order we execute them but not just
+        a sequentially
+        """
+        msg = "Called with wflow: %s"
+        self.logger.debug(msg, pformat(wflow))
+        # NOTE: The following dispatch logic is a subject to be changed at any time
+
+        # Do not clean any Resubmission, but still let them be archived
+        if wflow['RequestType'] == 'Resubmission':
+            wflow['ForceArchive'] = True
+            msg = "Skipping cleanup step for workflow: %s - RequestType is %s."
+            msg += "\nWill try to archive it directly."
+            self.logger.info(msg, wflow['RequestName'], wflow['RequestType'])
+        # Clean:
+        elif wflow['RequestStatus'] in ['rejected', 'aborted-completed']:
+            # NOTE: We do not check the ParentageResolved flag for these
+            #       workflows, but we do need to clean output data placement
+            #       rules from the agents for them
+            for pline in self.agentlines:
+                try:
+                    pline.run(wflow)
+                except Exception as ex:
+                    msg = "%s: General error from pipeline. Error: \n%s. "
+                    msg += "\nWill retry again in the next cycle."
```

Comment:
I'd suggest to keep this one in the same line as well. Shouldn't it include the workflow name too?

Reply:
Totally agree when it comes to the Workflow name. And disagree when it comes to including the General exception and the text in the same line. As you mentioned in the preovious comments it would be good to keep those on the same line, and i did so when it comes to a well defined messages, but when doing `str(ex)` it is basically a stack dump and it is multi line message anyway and not separating it on a new line just smears the main message with the stack dump.